### DISCUSSION TITLE
一段获取批里批里最高画质视频+音频的代码

### DISCUSSION THREAD

**Discussion by jc3213** [jc3213/download_with_aria2#37] [Ideas]
```javascript
(async function downloadBiliVideo() {
    var {aid, cid, title} = document.defaultView.__INITIAL_STATE__.videoData;
    var referer = location.href;
    var response = await fetch('https://api.bilibili.com/x/player/playurl?avid=' + aid + '&cid=' + cid + '&fnval=4050', {credentials: 'include'});
    var json = await response.json();
    var {video, audio} = json.data.dash;
    var vid = video[0].id;
    var array = [];
    video.forEach(json => {
        var {id, codecid, baseUrl} = json;
        if (id === vid) array[codecid] = baseUrl;
    });
    var viURI = array.pop();
    var auURI = audio[0].baseUrl;
    var message = {
        aria2c: 'Download With Aria2',
        download: {
            json: [
                { url: viURI, options: { out: title + '.video.mp4', referer } },
                { url: auURI, options: { out: title + '.audio.mp4', referer } }
            ]
        }
    };
    postMessage(message);
})();
```
![3](https://user-images.githubusercontent.com/8744936/205470881-32836955-3091-4b85-8060-2835f8c6858f.png)
![4](https://user-images.githubusercontent.com/8744936/205470883-343cbcc8-3870-4d90-b0be-3229f2843001.png)

**Comment by jc3213** [comment]
更改为 [28a8fd4](https://github.com/jc3213/download_with_aria2/commit/28a8fd49f6edccc3897bb222f27a7f8fcca51e95) 以后的结果 { `url`: `下载地址`, `options`: `设置` } 设置为Object对象

**Comment by jc3213** [comment]
因为[83642d7](https://github.com/jc3213/download_with_aria2/commit/83642d7524a40a715da71cf79ca565838380bc5b)已经支持直接通过`postMessage`交互，因此更改本脚本

**Reply by jc3213** [reply]
因为[37f4c64](https://github.com/jc3213/download_with_aria2/commit/37f4c64e964acd3d4614bcfe4682dcdd91009b6e)跟[fba8afb](https://github.com/jc3213/download_with_aria2/commit/fba8afbdd43e69fc9998f3b53218ba4229c2d600)改变了交互，再次更新脚本

**Comment by Zeoye** [comment]
您的邮件我已收到！如果需要回复的我会尽快回复！

**Comment by phinner** [comment]
:raised_eyebrow:
